rest/middleware.go

```diff
@@ -23,7 +42,7 @@ type evgAuthReadLogByIDMiddleware struct {
 // gimlet.Middleware that sends a http request to Evergreen to check if the
 // user is authorized to read the log they are trying to access based on the
 // given ID.
-func NewEvgAuthReadLogByIDMiddleware(sc data.Connector, evgConf *model.EvergreenConfig) gimlet.Middleware {
+func NewEvgAuthReadLogByIDMiddleware(sc data.Connector, evgConf *model.EvergreenConfig) *evgAuthReadLogByIDMiddleware {
```

Comment:
Does this function) and the one below it need to  be exported? It seems like they're only used within `rest`.